admin/app/helpers/solidus_admin/container_helper.rb

```diff
@@ -9,6 +9,9 @@ def container
     end
 
     def component(name)
+      # The dry namespace separator is not intended to be user-configurable.
+      name = name.tr('/', '.')
```

Comment:
We're not actually changing the container separator. But I think that's good for a couple of reasons:

- We're not actually mapping one-to-one filesystem with identifiers. E:g., we're skipping the `SolidusAdmin` namespace, and a path `components/solidus_admin/foo.rb` is registered as `components.foo`. Using `components/foo` would be confusing.
- The container is also used to manually register other dependencies with no many-to-one mapping to a file, like different instances of `SolidusAdmin::MainNavItem`. In that case, using something like `solidus_admin/main_nav_item/first_item` would also be confusing.

However, I think it's a great idea to hide all of that from within the `component` helper, as you did. But the comment and the commit titles are confusing as it looks as we did the opposite.

Reply:
> But the comment and the commit titles are confusing as it looks as we did the opposite.

Good call, I didn't recheck the title after discovering it was not configurable on DRY.rb, I'll update it 👍 

> We're not actually mapping one-to-one filesystem with identifiers. E:g., we're skipping the SolidusAdmin namespace, and a path components/solidus_admin/foo.rb is registered as components.foo. Using components/foo would be confusing.

To me rendering components is very similar to rendering partials, while the dots make me think of translations. It's clearly coming from what I'm used to in Rails, but I think that can be generalized to other Rails users. Even considering (comparatively rare) cases in which components are overridden making it look like a path is easily translatable to the file position in the file system.

> The container is also used to manually register other dependencies

With the code in this PR that is still the case (i.e. accessing using dots), but I think we would probably be better off with different spaces for data and components, be it a container, a configuration, or something else.

Reply:
Hey, @elia, we still forgot to remove the comment here. dry-system's separator is indeed configurable, but we're not changing it there but in the component helper (which makes sense as I pointed out above).

Reply:
Oh, ok, I didn't know that. Anyway, I'd remove the comment because of the reasons I detailed above. Non-blocking.